src/main/java/fr/esgi/al5_2/Tayarim/services/AuthService.java

```diff
@@ -3,87 +3,199 @@
-
-        String token = jwtHelper.generateToken(email, uuid);
-
-        return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
-        );
-
+    if (proprietaireDTO != null && !proprietaireService.verifyPassword(password,
+        proprietaireDTO.getId())) {
+      throw new ProprietaireNotFoundException();
+    } else if (administrateurDTO != null && !administrateurService.verifyPassword(password,
+        administrateurDTO.getId())) {
+      throw new AdministrateurNotFoundException();
     }
 
-    public AuthLoginResponseDTO authProprietaire(@NonNull String token){
-
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
-        
-        return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
-        );
-
+    String uuid = tokenCacheService.getFromCache(id);
+    if (uuid == null) {
+      uuid = UUID.randomUUID().toString();
+      tokenCacheService.addToCache(id, uuid);
     }
 
-    public void logoutProprietaire(@NonNull String token){
-
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
-
-        tokenCacheService.addToCache(proprietaireDTO.getId(), UUID.randomUUID().toString());
+    String token = jwtHelper.generateToken(email, uuid, isAdmin);
+
+    return new AuthLoginResponseDTO(id, token, isAdmin, nom, prenom, email, numTel);
+  }
+
+  /**
+   * Authentifie un utilisateur à partir d'un token.
+   *
+   * @param token Le token JWT fourni à l'utilisateur après une connexion réussie.
+   * @return {@link AuthResponseDTO}
+   * @throws TokenExpireOrInvalidException Si le token est expiré ou invalide.
+   */
+  public AuthResponseDTO auth(@NonNull String token) {
+
+    Entry<Long, Boolean> entry = verifyToken(token, false);
+
+    return new AuthResponseDTO(entry.getKey(), token, entry.getValue());
+
+  }
+
+  /**
+   * Déconnecte un utilisateur en invalidant le token actuel et en générant un nouveau.
+   *
+   * @param token Le token JWT de l'utilisateur à déconnecter.
+   */
+  public void logout(@NonNull String token) {
+
+    Entry<Long, Boolean> entry = verifyToken(token, false);
+
+    tokenCacheService.addToCache(entry.getKey(), UUID.randomUUID().toString());
+
+  }
+
+  /**
+   * Vérifie la validité du token JWT, en vérifiant également si l'utilisateur devrait être un
+   * administrateur.
+   *
+   * @param token         Le token JWT à vérifier.
+   * @param shouldBeAdmin Booléen indiquant si l'utilisateur doit être vérifié en tant
+   *                      qu'administrateur.
+   * @return {@link Entry}
+   * @throws TokenExpireOrInvalidException Si le token est invalide ou expiré, ou si l'utilisateur
+   *                                       n'est pas autorisé comme administrateur lorsque requis.
+   */
+  public Entry<Long, Boolean> verifyToken(@NonNull String token, boolean shouldBeAdmin) {
+    ProprietaireDTO proprietaireDTO;
+    AdministrateurDTO administrateurDTO;
+    Long id;
+    String email;
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.coding.VariableDeclarationUsageDistanceCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Distance between variable 'email' declaration and its first usage is 4, but allowed 3.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value).